lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskStatus.cs

```diff
@@ -246,12 +252,20 @@ public TaskStatusProto ToProto()
                 }
                 else if (_lastException.IsPresent())
                 {
+                    byte[] error;
+                    try
+                    {
+                        error = SerializeException(_lastException.Value);
+                    }
+                    catch (SerializationException)
+                    {
+                        error = SerializeException(new NonSerializableTaskException(_lastException.Value.ToString()));
```

Comment:
In my new iteration, I've opted to set the `InnerException` as the `SerializationException`, and `Exception.Message` as `Exception.ToString()` of the original Exception, which contains the original stack trace.